spring-cloud-dataflow-platform-kubernetes/src/test/java/org/springframework/cloud/dataflow/server/config/kubernetes/KubernetesPlatformPropertiesTests.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.config.kubernetes;
+
+import java.util.Map;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.deployer.spi.kubernetes.EntryPointStyle;
+import org.springframework.cloud.deployer.spi.kubernetes.ImagePullPolicy;
+import org.springframework.cloud.deployer.spi.kubernetes.KubernetesClientFactory;
+import org.springframework.cloud.deployer.spi.kubernetes.KubernetesDeployerProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Donovan Muller
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = KubernetesPlatformPropertiesTests.TestConfig.class)
+@ActiveProfiles("kubernetes-platform-properties")
+public class KubernetesPlatformPropertiesTests {
+
+	@Autowired
+	private KubernetesPlatformProperties kubernetesPlatformProperties;
+
+	@Test
+	public void deserializationTest() {
+		Map<String, KubernetesDeployerProperties> k8sAccounts = this.kubernetesPlatformProperties.getAccounts();
+		KubernetesClient devK8sClient = KubernetesClientFactory.getKubernetesClient(k8sAccounts.get("dev"));
+		KubernetesClient qaK8sClient = KubernetesClientFactory.getKubernetesClient(k8sAccounts.get("qa"));
+		assertThat(k8sAccounts).hasSize(2);
+		assertThat(k8sAccounts).containsKeys("dev", "qa");
+		assertThat(devK8sClient.getNamespace()).isEqualTo("dev1");
+		assertThat(devK8sClient.getMasterUrl().toString()).isEqualTo("http://192.168.0.1:8443");
+		assertThat(qaK8sClient.getMasterUrl().toString()).isEqualTo("http://192.168.0.2:8443");
+		assertThat(qaK8sClient.getNamespace()).isEqualTo("qaNamespace");
```

Comment:
this is currently failing:

```
[ERROR] Failures:
[ERROR]   KubernetesPlatformPropertiesTests.deserializationTest:58 expected:<"[qaNamespace]"> but was:<"[default]">
[INFO]
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
```

Reply:
@chrisjs Are you on the latest? It passed for me. Also, travis build looks green as well.

Reply:
@ilayaperumalg the latest commit i have is `53e54fa9cf5547e28394e133110d2092dab1cb37` in branch `GH-2627d` which seems to be the latest

Reply:
if for example you have a namespace specified in your kube config file for the particular context your using, that will get used. i put a further run down of my findings in the chat room. the following patch will set the namespace and the test will pass as it won't resolve from your config file, yet the yaml:

```
diff --git a/spring-cloud-dataflow-platform-kubernetes/src/test/resources/application-kubernetes-platform-properties.yml b/spring-cloud-dataflow-platform-kubernetes/src/test/resources/application-kubernetes-platform-properties.yml
index e33e6e28..8d72b085 100644
--- a/spring-cloud-dataflow-platform-kubernetes/src/test/resources/application-kubernetes-platform-properties.yml
+++ b/spring-cloud-dataflow-platform-kubernetes/src/test/resources/application-kubernetes-platform-properties.yml
@@ -17,6 +17,7 @@ spring:
               qa:
                 fabric8:
                   masterUrl: http://192.168.0.2:8443
+                  namespace: qaNamespace
                 namespace: qaNamespace
                 imagePullPolicy: IfNotPresent
                 entryPointStyle: boot
```